### PR TITLE
Make the output directories if not exists

### PR DESCRIPTION
**Pull request by rajbos**
I've bumped into this now three times: when the output directory does not exists (which it will not when the user starts with this), the action just crashes. I think it is a nicer experience to just create the directory for them: they indicate a path already where the files will be written to, so why not just make the dir for them as well? This PR handles that.

### CODE REVIEW

**Comment by rajbos** [comment]
Moved the mkdir code into the utils class. Not sure about the next step: do you want to add unit tests now or later? When now, I do need some help with the setup of overwriting the `fs` class then. 

I'd expect some tests for short/long file paths (multiple layers deep or not) as well as tests for when the folder already exists and when not?

File: src/utils.js [hunk 1]
```diff
@@ -58,12 +58,22 @@ const generateIssueContent = async (scores, renderBadge) => {
   return ejs.render(template, { scores: scoresInScope, renderBadge })
 }
 
+const makeDirectory = ({ path }) => {
```
Comment (UlisesGascon) [inline src/utils.js:61]: I believe that this should be a direct argument, as you are executing the function as `makeDirectory(scopePath)` in `action.js`

```suggestion
const makeDirectory = ( path ) => {
```

**Comment by UlisesGascon** [comment]
I can help you with the tests, I will add some commits to the PR :)

**Comment by UlisesGascon** [comment]
@rajbos I will merge this PR to the branch [feat/create-directories](https://github.com/UlisesGascon/openssf-scorecard-monitor/tree/feat/create-directories) and from that branch I will add the tests and create a new PR against `main`

**Pull request merged**